lib/engine/game/g_steam_over_holland/game.rb

```diff
@@ -225,20 +226,26 @@ def price_movement_chart
         end
 
         def sell_shares_and_change_price(bundle, allow_president_change: true, swap: nil, movement: nil)
-          super
-          num_shares = bundle.num_shares
-          unless bundle.owner == corporation.owner
-            # This allows for the ledges that prevent price drops unless the president is selling
+          shares_sold = bundle.num_shares
+          corporation = bundle.corporation
+          old_price = corporation.share_price
+          @share_pool.sell_shares(bundle, allow_president_change: allow_president_change, swap: swap)
+
+          if bundle.owner == corporation.owner
+            super
+          else
+            # This section allows for the ledges that prevent price drops unless the president is selling
             case corporation.share_price.type
             when :ignore_sale_unless_president
-              num_shares = 0
+              shares_sold = 0
             when :max_one_drop_unless_president
-              num_shares = 1
+              shares_sold = 1
             when :max_two_drops_unless_president
-              num_shares = 2 unless num_shares == 1
+              shares_sold = 2 unless shares_sold == 1
             end
+            shares_sold.times { @stock_market.move_left(corporation) }
```

Comment:
Let's rename this variable here, it's confusing to reassign shares_sold since it doesn't affect how many shares are sold, and its purpose here is stock market movement, so something referring to the market "moves" or "steps" would be better.